tests/utils/statefulset/statefulset.go

```diff
@@ -62,3 +62,15 @@ func RedefineWithLivenessProbe(statefulSet *v1.StatefulSet) *v1.StatefulSet {
 
 	return statefulSet
 }
+
+func RedefineWithContainerSpecs(statefulSet *v1.StatefulSet, containerSpecs []corev1.Container) *v1.StatefulSet {
```

Comment:
Do you need to return the pointer here at all?  It's already pass by reference.  Same with the `RedefineWithReplicaNumber` func?

Reply:
Well, that's the standard RedefineXX behavior for all these helpers, so you can do something like:
```
dep := RedefineDeploymentWithY(RedefineDeploymentWithX(DefineDeployment(x,y,z)))
```